ON-15183: Update README.md (#69)
Xilinx-CNS/kubernetes-onload: README.md

scripts/onload_mkcontainer --user kitchen-sink.kube.test:5000/onload-user:latest *.tgz
```
6. Push the Onload userland image:
```
docker push kitchen-sink.kube.test:5000/onload-user:latest
```
By default, the Onload userland uses UBI libc. Please check and patch the Dockerfile if this is incompatible with your application's environment.

7. Check out this branch of this repository.
8. Build an Onload kernel image:
```
make -C build/onload-module/ onload-module ONLOAD_SOURCE_IMAGE_REPO=kitchen-sink.kube.test:5000/onload-source ONLOAD_SOURCE_IMAGE_TAG=latest ONLOAD_MODULE_IMAGE_REPO=kitchen-sink.kube.test:5000/onload-module
```
However, since we default to OpenShift, please also pass the correct `DTK_AUTO` and `KERNEL_VERSION` to build (and possibly cross-compile) the Onload and SFC kernel modules. I suggest examining Makefile.

9. Push the Onload kernel image (I copied the autogenerated hash and kernel version):
```
docker push kitchen-sink.kube.test:5000/onload-module:deccdb8d036a4c7794dea2cda106b3c112b374a9-4.18.0-372.49.1.el8_6.x86_64
```

### Prepare Kubernetes cluster

The Onload Operator v3 depends on third-party software:

1. Kernel Module Manamagent (KMM) Operator v1.1.1.

> 2. CNI that supports ipvlan or macvlan configurations. The Onload Operator currently suggests Multus CNI. A sample configuration is https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/examples/macvlan-pod.yml.

Reviewer: I don't think we are actually tied to `ipvlan` or `macvlan`. We should be able to use multus in general (although we have only tested with ipvlan/macvlan)
```suggestion
2. Multus CNI. A sample configuration using macvlan is https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/examples/macvlan-pod.yml.
```

Author: Thanks, applied. Also, added a note that `NetworkAttachmentDefinition` is used later.

@@ -55,9 +55,9 @@ spec:
     # Example image locations using openshift local image registry.
     kernelMappings:
       - regexp: '^.*\.x86_64$'
-        kernelModuleImage: image-registry.openshift-image-registry.svc:5000/onload-clusterlocal/onload-module:v8.1.0-${KERNEL_FULL_VERSION}
+        kernelModuleImage: kitchen-sink.kube.test:5000/onload-module:deccdb8d036a4c7794dea2cda106b3c112b374a9-4.18.0-372.49.1.el8_6.x86_64
         sfc: {}
-    userImage: image-registry.openshift-image-registry.svc:5000/onload-clusterlocal/onload-user:v8.1.0
+    userImage: kitchen-sink.kube.test:5000/onload-user:latest
     version: 8.1.0
     imagePullPolicy: Always
   devicePlugin:
```
Please also make sure `devicePluginImage` is correct. Another important field is the node `selector`, which tells the Onload Operator where to deploy the Onload Device Plugin DaemonSet and Module kind (KMM).

5. Finally, deploy the Onload CR:
```
oc apply -k config/samples/
```

## Run Onloaded application

At this point, Onload is deployed at the cluster, and the users can run Onloaded applications, e.g.
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test
  annotations:

>     k8s.v1.cni.cncf.io/networks: ipvlan-sf0

Reviewer: Could you add a note/comment for this line saying that this should be the name of the NetworkAttachmentDefinition that they created earlier (and perhaps make it obvious that they need to create the NetworkAttachmentDefinition, just having the CNI isn't sufficient).
